.docker/release/Dockerfile

```diff
@@ -1,51 +1,116 @@
-    docker-php-ext-configure ldap --with-libdir=lib/$(uname -m)-linux-gnu/ && \
-    docker-php-ext-configure intl && \
-    docker-php-ext-configure gd --with-freetype --with-jpeg && \
-    docker-php-ext-install ldap opcache pdo_mysql pdo_pgsql zip intl gd && \
-    php -m
+# gd
+RUN set -eux; \
+    apk add --no-cache freetype libjpeg-turbo libpng; \
+    apk add --no-cache --virtual .deps freetype-dev libjpeg-turbo-dev libpng-dev; \
+    docker-php-ext-configure gd --with-freetype --with-jpeg; \
+    docker-php-ext-install gd; \
+    apk del .deps
+# gnupg
+RUN set -eux; \
+    apk add --no-cache gnupg gpgme; \
+    apk add --no-cache --virtual .deps gpgme-dev; \
+    apk add --no-cache --virtual .build-dependencies $PHPIZE_DEPS; \
+    pecl install gnupg; \
+    docker-php-ext-enable gnupg; \
+    docker-php-source delete; \
+    apk del .build-dependencies; \
+    apk del .deps
+# intl
+RUN set -eux; \
+    apk add --no-cache icu-libs; \
+    apk add --no-cache --virtual .deps icu-dev; \
+    docker-php-ext-configure intl; \
+    docker-php-ext-install intl; \
+    apk del .deps
+# ldap
+RUN set -eux; \
+    apk add --no-cache libldap; \
+    apk add --no-cache --virtual .deps openldap-dev; \
+    docker-php-ext-configure ldap; \
+    docker-php-ext-install ldap; \
+    apk del .deps
+# mysql
+RUN docker-php-ext-install pdo_mysql
+# opcache
+RUN docker-php-ext-install opcache
+# postgres
+RUN set -eux; \
+    apk add --no-cache postgresql-libs; \
+    apk add --no-cache --virtual .deps postgresql-dev; \
+    docker-php-ext-install pdo_pgsql; \
+    apk del .deps
+# redis
+RUN set -eux; \
+    apk add --no-cache liblzf zstd-libs; \
+    apk add --no-cache --virtual .deps zstd-dev; \
+    apk add --no-cache --virtual .build-dependencies $PHPIZE_DEPS; \
+    pecl install igbinary; \
+    docker-php-ext-enable igbinary; \
+    pecl install --configureoptions 'enable-redis-igbinary="yes" enable-redis-lzf="yes" enable-redis-zstd="yes"' redis; \
+    docker-php-ext-enable redis; \
+    docker-php-source delete; \
+    apk del .build-dependencies; \
+    apk del .deps
+# zip
+RUN set -eux; \
+    apk add --no-cache libzip; \
+    apk add --no-cache --virtual .deps libzip-dev; \
+    docker-php-ext-install zip; \
+    apk del .deps
 
 # Install snappymail
-WORKDIR /tmp
-COPY ${FILES_ZIP} .
-RUN mkdir /snappymail && \
-    unzip -q ${FILES_ZIP} -d /snappymail && \
-    find /snappymail -type d -exec chmod 755 {} \; && \
-    find /snappymail -type f -exec chmod 644 {} \; && \
-    rm -rf ${FILES_ZIP}
+# The 'www-data' user/group in alpine is 82:82. The 'nginx' user/group in alpine is 101:101, and is part of www-data group
+COPY --chown=www-data:www-data --from=builder /snappymail /snappymail
+# Use a custom snappymail data folder
+RUN mv -v /snappymail/data /var/lib/snappymail; \
+    ( \
+        echo "<?php"; \
+        echo "define('APP_DATA_FOLDER_PATH', '/var/lib/snappymail/');"; \
+        echo "?>"; \
+    ) > /snappymail/include.php;
```

Comment:
Patching file from dockerfile looks quite confusing. Maybe it is possible to copy whole `/snappymail/include.php` with prepared context?

Reply:
Yes. I'll do this after I removed `logrotate` and `tail -f`.